eth/catalyst/api.go

```diff
@@ -639,7 +827,69 @@ func (api *ConsensusAPI) newPayload(params engine.ExecutableData, versionedHashe
 		return api.invalid(err, parent.Header()), nil
 	}
 	hash := block.Hash()
-	return engine.PayloadStatusV1{Status: engine.VALID, LatestValidHash: &hash}, nil
+
+	// If witness collection was requested, inject that into the result too
+	var sw *engine.StatelessWitnessV1
+	if proofs != nil {
+		sw = proofs.ToStatelessWitnessV1()
+	}
+	return engine.PayloadStatusV1{Status: engine.VALID, Witness: sw, LatestValidHash: &hash}, nil
+}
+
+func (api *ConsensusAPI) executeStatelessPayload(params engine.ExecutableData, versionedHashes []common.Hash, beaconRoot *common.Hash, witness engine.StatelessWitnessV1) (engine.StatelessPayloadStatusV1, error) {
+	log.Trace("Engine API request received", "method", "ExecuteStatelessPayload", "number", params.Number, "hash", params.BlockHash)
+
+	block, err := engine.ExecutableDataToBlockNoHash(params, versionedHashes, beaconRoot)
+	if err != nil {
+		bgu := "nil"
+		if params.BlobGasUsed != nil {
+			bgu = strconv.Itoa(int(*params.BlobGasUsed))
+		}
+		ebg := "nil"
+		if params.BlobGasUsed != nil {
```

Comment:
`if params.ExcessBlobGas != nil`

Reply:
the same error in `newPayload`, please also fix it

Reply:
Ah, it was broken in the original code, that's how it got there. Seems to be fixed upstream.